core/quantity/src/Formatter/FormatterSpec.ts

```diff
@@ -60,6 +61,60 @@ export class FormatterSpec {
   public get azimuthBaseConversion(): UnitConversionProps | undefined { return this._azimuthBaseConversion; }
   public get revolutionConversion(): UnitConversionProps | undefined { return this._revolutionConversion; }
 
+  /** Build conversion specs for ratio format with explicit numerator/denominator units. */
+  private static async getRatioUnitConversions(format: Format, unitsProvider: UnitsProvider, persistenceUnit: UnitProps): Promise<UnitConversionSpec[]> {
+    const conversions: UnitConversionSpec[] = [];
+
+    // Already validated by caller that hasRatioUnits is true
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const [numeratorUnit, numeratorLabel] = format.ratioUnits![0];
```

Comment:
Since `format` isn't used elsewhere in this method, and you're sure that format.ratioUnits[0] and ...[1] are present, just pass in a tuple with the two ratioUnits? Or each as a separate arg? Then no non-null assertion would be required.